cmd/provider/main.go

```diff
@@ -47,6 +47,7 @@ func main() {
 		app              = kingpin.New(filepath.Base(os.Args[0]), "Terraform based Crossplane provider for Azuread").DefaultEnvars()
 		debug            = app.Flag("debug", "Run with debug logging.").Short('d').Bool()
 		syncPeriod       = app.Flag("sync", "Controller manager sync period such as 300ms, 1.5h, or 2h45m").Short('s').Default("1h").Duration()
+		pollInterval     = app.Flag("poll", "Poll interval controls how often an individual resource should be checked for drift.").Default("10m").Duration()
```

Comment:
Let's also add the [`--max-reconcile-rate` option](https://github.com/upbound/provider-aws/blob/025f49bfecf5c4a1bcd5ac2cb9cb6de007be70ea/cmd/provider/main.go#L43) here, and pass them to the crossplane-runtime controller options.